src/config/config.service.ts

```diff
@@ -154,6 +165,38 @@ export class ConfigService {
         return this.ambsConfig.get(amb)?.globalProperties[key];
     }
 
+    private async validateChains(): Promise<void> {
+        const handleError = (chainId: string) => {
+            throw new Error(`Error validating the Chain ID for chain ${chainId}`);
+        };
+
+        const validateChainIdFromRPC = async (rpc: string, expectedChainId: string): Promise<boolean> => {
+            const provider = new JsonRpcProvider(rpc, undefined, { staticNetwork: true });
+            try {
+                const network = await provider.getNetwork();
+                const actualChainId = network.chainId.toString();
+                return actualChainId === expectedChainId;
+            } catch (error) {
+                return false;
+            }
+        };
+
+        const validationPromises = this.rawConfig['chains'].map(async (rawChainConfig: any) => {
```

Comment:
Iterate over the `this.chainsConfig` map instead of `this.rawConfig`, as that contains the parsed and validated configuration (increase the separation of concerns).